### ISSUE TITLE
p:chart zoom with date/time axis shows nothing

### ISSUE THREAD

**Issue by douglasjunior** [primefaces/primefaces#1736]
I have problem to set the zoom on a chart with "date time" axis. I made a simulation comparing the chart of Primefaces with native jqplot. With [tag:jqplot] native works fine, but with [tag:primefaces] not.

It seems to me an bug in the calculation of the viewport.

![Gif sample](https://raw.githubusercontent.com/douglasjunior/PrimefacesChartZoomTest/master/demo.gif)

My JSF page:

``` html
    <p:chart type="line" model="#{chartController.model}" id="chart" style="height: 400px" />
```

My managed bean:

``` java
    @Named(value = "chartController")
    @ViewScoped
    public class ChartController implements Serializable {

        private LineChartModel model;

        public ChartController() {
        }

        @PostConstruct
        public void init() {
            long[][] lines = {{1334856823000l, 2}, {1334856853000l, 1}, {1334856883000l, 0}, {1334856913000l, 4}, {1334856914000l, 13},
            {1334856943000l, 16}, {1334856973000l, 23}, {1334857003000l, 24}, {1334857033000l, 36}, {1334857063000l, 14}, {1334857093000l, 1}};

            model = new LineChartModel();
            model.setTitle("Primefaces Chart");
            model.setZoom(true);

            LineChartSeries series = new LineChartSeries();

            for (long[] line : lines) {
                series.set(line[0], line[1]);
            }

            DateAxis xaxis = new DateAxis();
            xaxis.setTickFormat("%e/%b %H:%M");
            xaxis.setTickAngle(-30);
            xaxis.setMin(1334856823000l); // if not set this, chart not work
            model.getAxes().put(AxisType.X, xaxis);

            Axis yaxis = new LinearAxis();
            yaxis.setMin(0);
            model.getAxes().put(AxisType.Y, yaxis);

            model.addSeries(series);
        }

        public LineChartModel getModel() {
            return model;
        }

    }
```

My jqplot native code:

``` html
    <div id="chart" style="height: 400px"></div>

    <script>

        $(document).ready(function () {
            $.jqplot.config.enablePlugins = true;

            var lines = [[1334856823000, 2], [1334856853000, 1], [1334856883000, 0], [1334856913000, 4], [1334856914000, 13],
                [1334856943000, 16], [1334856973000, 23], [1334857003000, 24], [1334857033000, 36], [1334857063000, 14], [1334857093000, 1]];

            $.jqplot('chart', [lines], {
                title: "Jqplot Native",
                axes: {
                    xaxis: {
                        renderer: $.jqplot.DateAxisRenderer,
                        tickRenderer: $.jqplot.CanvasAxisTickRenderer,
                        tickOptions: {
                            formatString: '%e/%b %H:%M',
                            angle: -30
                        }
                    },
                    yaxis: {
                        renderer: $.jqplot.LinearAxisRenderer,
                        tickRenderer: $.jqplot.CanvasAxisTickRenderer,
                        min: 0
                    }
                },
                cursor: {zoom: true}
            });

        });

    </script>
```

The sample project is: https://github.com/douglasjunior/PrimefacesChartZoomTest
- Primefaces: 5.3
- Java EE: 7.0
- GlassFish: 4.1.1

**# UPDATE 2016-08-03:**

Same problem with Primefaces 6.0

**# UPDATE 2016-09-01:**

Based on **lalitha ramakrishnan** [answer](http://stackoverflow.com/a/39264457/2826279) I make it work just including the `jqplot.dateAxisRenderer.min.js` file in xhtml page. For some bug, the Primefaces dont includes this automatically.

Now the lines was always `smooth=true`. I tried configure by setting `lineSeries.setSmoothLine(false)` or with `extender`, but not worked.

Original question: http://stackoverflow.com/questions/36864806/pchart-zoom-with-date-time-axis-shows-nothing/39264457#39264457

**Comment by melloware** [comment]
Can you test this again?  From looking at the 6.1 code on it seems.

1. PF fixed the line Smooth bug:  https://github.com/primefaces/primefaces/issues/1608 in 6.0.2

2. The jqplot.dateAxisRenderer.js is included by PF in the charts plugin?

**Comment by faxe1008** [comment]
Is there any progress made on this? I can replicate the issue in version 6.2

**Comment by melloware** [comment]
@faxe1008 @douglasjunior Can you guys try again with my attached test project?  I can't reproduce your issue using PF 6.3-SNAPSHOT.  The DateAxisRenderer is included in the JS file by default.

[pf-1736.zip](https://github.com/primefaces/primefaces/files/2680403/pf-1736.zip)

TO run this just run "mvn clean jetty:run" and navigate to http://localhost:8080/primefaces-test/test.xhtml

You must first build PF  6.3-SNAPSHOT ..

```
git clone https://github.com/primefaces/primefaces.git
cd primefaces
mvn clean install
```

**Comment by melloware** [comment]
@tandraschko @Rapster Safe to mark this ticket as fixed for 6.3 and close it.  I have created a test case that proves its fixed.
